tests/test_utils_mean.py

```diff
@@ -1,134 +1,127 @@
-    assert_array_equal(C, np.eye(3))
-
-
-def test_logdet_mean():
-    """Test the logdet mean"""
-    covmats, _, _ = generate_cov(100, 3)
-    C = mean_logdet(covmats)
-    assert C.shape == (3, 3)
-
-
-def test_logdet_mean_with_init():
-    """Test the logdet mean"""
-    covmats, _, _ = generate_cov(100, 3)
-    C = mean_logdet(covmats, init=covmats[0])
-    assert C.shape == (3, 3)
+    assert np.all(C == np.eye(3))
 
 
-def test_ald_mean():
-    """Test the Ale mean"""
-    covmats, _, _ = generate_cov(100, 3)
-    C = mean_ale(covmats)
-    assert C.shape == (3, 3)
+def test_alm_mean():
+    """Test the ALM mean"""
+    covmats, _, _ = generate_cov(3, 3)
+    C_alm = mean_alm(covmats)
+    C_riem = mean_riemann(covmats)
+    assert C_alm == approx(C_riem)
 
 
-def test_kullback_mean():
-    """Test the kullback mean"""
-    covmats, _, _ = generate_cov(100, 3)
-    C = mean_kullback_sym(covmats)
+def test_alm_mean_maxiter():
+    """Test the ALM mean with max iteration"""
+    covmats, _, _ = generate_cov(3, 3)
+    C = mean_alm(covmats, maxiter=1, verbose=True)  # maxiter reached
     assert C.shape == (3, 3)
 
 
-def test_harmonic_mean():
-    """Test the harmonic mean"""
-    covmats, _, _ = generate_cov(100, 3)
-    C = mean_harmonic(covmats)
-    assert C.shape == (3, 3)
-
-
-def test_wasserstein_mean():
-    """Test the wasserstein mean"""
-    covmats, _, _ = generate_cov(100, 3)
-    C = mean_wasserstein(covmats)
-    assert C.shape == (3, 3)
-
-
-def test_mean_covariance_riemann():
-    """Test mean_covariance for riemannian metric"""
-    covmats, _, _ = generate_cov(100, 3)
-    C = mean_covariance(covmats, metric='riemann')
-    Ctrue = mean_riemann(covmats)
-    assert_array_equal(C, Ctrue)
-
-
-def test_mean_covariance_logdet():
-    """Test mean_covariance for logdet metric"""
-    covmats, _, _ = generate_cov(100, 3)
-    C = mean_covariance(covmats, metric='logdet')
-    Ctrue = mean_logdet(covmats)
-    assert_array_equal(C, Ctrue)
-
-
-def test_mean_covariance_logeuclid():
-    """Test mean_covariance for logeuclid metric"""
-    covmats, _, _ = generate_cov(100, 3)
-    C = mean_covariance(covmats, metric='logeuclid')
-    Ctrue = mean_logeuclid(covmats)
-    assert_array_equal(C, Ctrue)
-
-
-def test_mean_covariance_euclid():
-    """Test mean_covariance for euclidean metric"""
-    covmats, diags, A = generate_cov(100, 3)
-    C = mean_covariance(covmats, metric='euclid')
-    Ctrue = mean_euclid(covmats)
-    assert_array_equal(C, Ctrue)
+def test_alm_mean_2trials():
+    """Test the ALM mean with 2 trials"""
+    covmats, _, _ = generate_cov(2, 3)
+    C = mean_alm(covmats)  # n_trials=2
+    assert np.all(C == geodesic_riemann(covmats[0], covmats[1], alpha=0.5))
+
+
+@pytest.mark.parametrize(
+    "metric,mean",
+    [
+        ("riemann", mean_riemann),
+        ("logdet", mean_logdet),
+        ("logeuclid", mean_logeuclid),
+        ("euclid", mean_euclid),
+        ("alm", mean_alm),
```

Comment:
Why other metrics/means ('identity', 'wasserstein', 'ale', 'harmonic', 'kullback_sym') are not tested here?

Reply:
yes, I just refactored existing tests, but we should tests these ones also!